controllers/cinder_controller.go

```diff
@@ -945,6 +963,26 @@ func (r *CinderReconciler) backupDeploymentCreateOrUpdate(ctx context.Context, i
 	return deployment, op, err
 }
 
+func (r *CinderReconciler) backupCleanupDeployment(ctx context.Context, instance *cinderv1beta1.Cinder) error {
+	deployment := &cinderv1beta1.CinderBackup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-backup", instance.Name),
+			Namespace: instance.Namespace,
+		},
+	}
+	key := client.ObjectKeyFromObject(deployment)
+	err := r.Client.Get(ctx, key, deployment)
+	if err == nil && cinder.GetOwningCinderName(deployment) == instance.Name {
```

Comment:
- I think we should log a warning if we find the object but we don't own it.
- Shouldn't we check that the error is not found and not a different error?

Reply:
I can definitely expand things to add addtional log message for various error conditions. The two that come to mind are:

1. Any error other than "not found"
2. The deployment isn't owned by us

Reply:
Actually, I think the whole "not owned by us" it extraneous when the team agreed there would be only one cinder CR per namespace. I'm going to push an update that updates the logging, but it will also use lib-common code to look for a backup deployment in the given namespace.